src/main/java/com/pgs/booking/service/WeatherService.java

```diff
@@ -0,0 +1,72 @@
+package com.pgs.booking.service;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.pgs.booking.model.dto.WeatherDto;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+@Getter
+@Setter
+@Service
+@RequiredArgsConstructor
+public class WeatherService {
+
+    private final String hostName = "x-rapidapi-host";
+    private final String hostValue = "community-open-weather-map.p.rapidapi.com";
+    private final String keyName = "x-rapidapi-key";
+    private final String keyValue = "c72dd541f1msh23890d6e014e604p10d3c9jsn3a67da9a0d84";
+
+    public WeatherDto getWeatherByCity(String city) {
+        OkHttpClient client = new OkHttpClient();
+        String urlWeather = "https://community-open-weather-map.p.rapidapi.com/weather?q=" + city + "&units=imperial&mode=JSON";
```

Comment:
FYI you have used rapidapi not directly 
api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key}

but ok

Reply:
I have corrected and use it directly
api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key}